### PR TITLE
use common stackManager interface and fakes across packages

### PR DESCRIPTION
**Pull request by aclevername**
### Description

First of many small PRs relating to #2931. This introduces a stackManager interface and fake that can be used to mock out the package directly, it should make testing this in the future easier.

This also highlights how INSANELY big the stack collection is, we should try to refactor this out into smaller sub packages in the future

### Checklist
- [ ] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [ ] Manually tested
- [ ] Made sure the title of the PR is a good description that can go into the release notes
- [ ] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

File: pkg/actions/addon/addon.go [hunk 1]
```diff
@@ -22,19 +20,11 @@ type Manager struct {
 	clusterProvider *eks.ClusterProvider
 	withOIDC        bool
 	oidcManager     *iamoidc.OpenIDConnectManager
-	stackManager    StackManager
+	stackManager    manager.StackManager
 	clientSet       kubeclient.Interface
 }
 
-//go:generate counterfeiter -o fakes/fake_stack_manager.go . StackManager
-type StackManager interface {
```
Comment (michaelbeaumont) [inline pkg/actions/addon/addon.go:30]: These custom interfaces are actually kind of useful for seeing which parts of the `StackManager` are used around the codebase, could be a guide to splitting it up

**Review by michaelbeaumont — APPROVED** [review]

File: pkg/actions/addon/addon.go (hunk 1 above)
Comment (aclevername) [reply pkg/actions/addon/addon.go:30]: Yeah I don't disagree, me and @Callisto13 had a chat about wether to keep these or not in the issue https://github.com/weaveworks/eksctl/issues/2931#issuecomment-748109410, I'm quite torn.

Comment (Callisto13) [reply pkg/actions/addon/addon.go:30]: Mike does have a point that it could make things easier when we come to splitting that beast of a stack manager.

Comment (aclevername) [reply pkg/actions/addon/addon.go:30]: It does, but like you pointed out in the issue it means loads of duplication in what is likely every package and if I had to guess would make people writing tests less likely 🤷  I'm leaning towards this at the moment as it enables everyone to very quick start adding tests

**Pull request merged**